packages/front-end/components/Radix/RadioGroup.tsx

```diff
@@ -0,0 +1,89 @@
+import { ReactElement } from "react";
+import { Flex, Text, RadioGroup as RadixRadioGroup } from "@radix-ui/themes";
+import { MarginProps } from "@radix-ui/themes/dist/cjs/props/margin.props";
+import HelperText, { getRadixColor } from "@/components/Radix/HelperText";
+
+export type RadioOptions = {
+  value: string;
+  label?: string | ReactElement;
+  description?: string | ReactElement;
+  sub?: string;
+  error?: string;
+  errorLevel?: "error" | "warning";
+  disabled?: boolean;
+}[];
+
+export type Props = {
+  disabled?: boolean;
+  options: RadioOptions;
+  value: string;
+  setValue: (value: string) => void;
+} & MarginProps;
+
+export default function RadioGroup({
+  options,
+  value,
+  setValue,
+  disabled,
+  ...containerProps
+}: Props) {
+  // get color for selected option
+  const selectedOption = options.find((o) => o.value === value);
+  const radioColor = selectedOption?.error
+    ? getRadixColor(selectedOption?.errorLevel ?? "error")
+    : "violet";
+
+  return (
+    <Flex {...containerProps}>
+      <Text as="label" size="2" color={disabled ? "gray" : undefined}>
+        <RadixRadioGroup.Root
+          value={value}
+          onValueChange={(val) => setValue(val)}
+          disabled={disabled}
+          color={radioColor}
+        >
+          {options.map(
+            ({
+              value,
+              label,
+              description,
+              sub,
+              disabled,
+              error,
+              errorLevel = "error",
+            }) => {
+              return (
+                <RadixRadioGroup.Item
+                  key={value}
+                  value={value}
+                  disabled={disabled}
+                >
+                  <Text color={disabled ? "gray" : undefined}>
+                    <Flex direction="column" gap="1">
+                      <Text weight="bold">{label || value}</Text>
+                      <Flex direction="column" gap="1">
+                        {description && (
+                          <>
+                            <Text weight="regular">{description}</Text>
+                            {sub && (
+                              <Text size="1" weight="light">
+                                {sub}
+                              </Text>
+                            )}
+                          </>
+                        )}
+                        {error && (
+                          <HelperText status={errorLevel}>{error}</HelperText>
+                        )}
```

Comment:
It looks like in the Figma, the HelperText is only shown when the option is selected.  Also might need to tweak how you're specifying color.  Instead of doing it at the top-level, can we do it at the individual radio item level?

![image](https://github.com/user-attachments/assets/83874e7d-36ff-4461-963a-512e6cbfb44e)

Reply:
>can we do it at the individual radio item level?

I looked at this, and there isn't a prop to set it that way. I'll try and do it manually.